src/go/cmd/update-collection-v3/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	inFileFlag  = flag.String("in", "values.yaml", "input values.yaml to be migrated")
+	outFileFlag = flag.String("out", "new_values.yaml", "output values.yaml")
+)
+
+func main() {
+	flag.Parse()
+
+	valuesV2, err := parseValues(*inFileFlag)
+	if err != nil {
+		log.Fatalf("failed reading %s: %v", *inFileFlag, err)
+	}
+
+	if err := migrate(valuesV2); err != nil {
+		log.Fatalf("failed migrating %s: %v", *inFileFlag, err)
+	}
+
+	if err := toYaml(valuesV2, *outFileFlag); err != nil {
+		log.Fatalf("failed writing %s: %v", *outFileFlag, err)
+	}
+}
+
+func parseValues(path string) (ValuesV2, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return ValuesV2{}, fmt.Errorf("cannot open file %s: %w", path, err)
+	}
+
+	decoder := yaml.NewDecoder(bufio.NewReader(f))
+
+	var valuesV2 ValuesV2
+	if err := decoder.Decode(&valuesV2); err != nil {
+		return ValuesV2{}, fmt.Errorf("cannot unmarshal data from %s: %w", path, err)
+	}
+
+	return valuesV2, nil
+}
```

Comment:
What happens for partial `values`yaml` or for `values.yaml` with additional keys?

Reply:
What do you mean by partial?

Additional ( if I understand it correctly by not defined in the struct) ones will be dropped. Maybe this could be worked around using `yaml.Node`

Reply:
> What do you mean by partial?

with missing keys, e.g.:
```yaml
sumologic:
  accessId: xxx
```

Reply:
There is nothing missing in the yaml snippet you posted. There is no notion of "required" in this package.

Reply:
my doubt is, that Stringifying `ValuesV2` will override defaults with empty strings